app/scripts/modules/core/src/managed/resources/useGetResourceLink.hook.ts

```diff
@@ -0,0 +1,18 @@
+import { useSref } from '@uirouter/react';
+
+import { IResourceLinkProps, resourceManager } from './resourceRegistry';
+
+export const useGenerateLink = (props: IResourceLinkProps): { href: string; isExternal?: boolean } | undefined => {
+  const externalLink = resourceManager.getExperimentalDisplayLink(props);
+  const routingInfo = resourceManager.getNativeResourceRoutingInfo(props) ?? { state: '', params: {} };
+  const routeProps = useSref(routingInfo.state, routingInfo.params);
+
+  if (externalLink) {
+    return { href: externalLink, isExternal: true };
+  }
+
+  if (routeProps.href) {
+    return { href: routeProps.href };
+  }
```

Comment:
use the `{ onClick }` handler returned from `useSref`

Reply:
Why? use href is better for accessibility when possible

Reply:
Use both the href and the onClick handler.  All the href behaviors are retained (link hover, ctrl-click opens in new tab, etc).  Using the onClick handler is a safer way to initiate navigation with the router.  It's also the pattern _all the other code in deck_ uses.

Reply:
oh, do you add both to the component and preventDefault in the onClick?

Reply:
yep, the onClick handler it returns handles preventDefault correctly, as well as handling shit/ctrl/alt/etc clicks